crates/interpreter/src/valid.rs

```diff
@@ -427,9 +427,12 @@ impl SideTable {
     fn stitch(&mut self, source: SideTableBranch, target: SideTableBranch) -> CheckResult {
         let delta_ip = Self::delta(source, target, |x| x.parser.as_ptr() as isize)?;
         let delta_stp = Self::delta(source, target, |x| x.side_table as isize)?;
-        // TODO(dev/fast-interp): Compute the fields below.
-        let val_cnt = 0;
-        let pop_cnt = 0;
+        let val_cnt = u32::try_from(source.result).map_err(|_| {
```

Comment:
I think it's actually better to have `val_cnt` stored in the branch target instead, since that's the one defining it. This should simplify `br_label()`.